submariner/cable-driver-protocols-enhancement.md

```diff
@@ -0,0 +1,76 @@
+# Cable Driver Enhancement - IP-in-IP
+
+## Summary
+
+Submariner today supports three cable drivers: VXLAN, IPSec (tunnel mode) and WireGuard. The
+goal of this proposal is to outline additional enhancements needed for the Submariner Cable Driver
+to support of IP-in-IP for unencrypted connections between clusters or for use as an overlay with
+IPSec in transport mode for encrypted connections between clusters.
```

Comment:
Is this EP only cover IPinIP as un-encrypted cable driver or also as an overlay with IPSec transport mode?

Reply:
I have the same question. I see some potential draw-backs with IP-in-IP as a cable driver.
1. Unlike VxLAN cable driver, I'm not sure if we can use IP-in-IP cable driver along with Load-balancer mode.
2. IP-in-IP cable driver cannot work in environments where there is NAT traversal without explicit 1:1 NAT. 

The above points IMHO are important ones to be considered before adding a new cable driver in the code. 

So the question is, should IP-in-IP considered only as an internal overlay tunnel with IPsec transport mode? Even if its the case, can we just use VxLAN itself both as an independent Cable driver as well as overlay tunnel with IPsec transport mode?

Reply:
The load balancer supports need to be checked as Sridhar mentioned since it is planned to be the default mode going forward.

Reply:
> Re the Load-balancer mode, I guess it depends on the meaning of Load-balancer mode, is it Submariner's Load-balancer mode where it's getting a public IP or is it Load-balancer mode from the POV of an L4 Load Balancer sitting infront of a number of Submariner GWs and distributing packets across them?

When Submariner is configured with Loadbalancer mode, we create a standard K8s Service of type Loadbalancer and configure the ports with UDP/4500 (so that IPsec tunnels or vxLAN tunnels can be setup). So its a L4 load balancer.

> But yes, I take your point about VXLAN being usable as both an independent driver as well as an overlay over IPSec transport mode..

I'd like to hear the views from others in the team whether we should add an additional IP-in-IP cable-driver since VxLAN can be used for both use-cases. Basically, having additional driver also implies maintenance/support/CI overhead, so I'm thinking on those lines.

Reply:
For L4 loadbalancer this will be an issue for sure, if the balancer is round robin probably not as much of an issue...